dtfs-central-api/src/repositories/tfm-deals-repo/tfm-deal-cancellation.repo.submitDealCancellation.test.ts

```diff
@@ -0,0 +1,109 @@
+import { DEAL_SUBMISSION_TYPE, DealNotFoundError, InvalidDealIdError, MONGO_DB_COLLECTIONS, TFM_DEAL_STAGE } from '@ukef/dtfs2-common';
+import { generateAuditDatabaseRecordFromAuditDetails, generateTfmAuditDetails } from '@ukef/dtfs2-common/change-stream';
+import { ObjectId } from 'mongodb';
+import { flatten } from 'mongo-dot-notation';
+import { getUnixTime } from 'date-fns';
+import { mongoDbClient as db } from '../../drivers/db-client';
+import { TfmDealCancellationRepo } from './tfm-deal-cancellation.repo';
+import { aTfmUser } from '../../../test-helpers';
+
+const dealId = new ObjectId();
+
+const mockBankRequestDate = getUnixTime(new Date('2024-01-01'));
+const mockEffectiveFrom = getUnixTime(new Date('2025-02-02'));
+const mockReason = 'mock reason';
+
+const mockDealCancellationObject = { reason: mockReason, bankRequestDate: mockBankRequestDate, effectiveFrom: mockEffectiveFrom };
+
+const tfmUserId = aTfmUser()._id;
+
+const auditDetails = generateTfmAuditDetails(tfmUserId);
+
+describe('tfm-deals-cancellation-repo', () => {
+  const updateOneMock = jest.fn();
+  const getCollectionMock = jest.fn();
+
+  beforeAll(() => {
+    jest.useFakeTimers().setSystemTime(1727876437063);
```

Comment:
surprised to see the repo caring about the system time - is this needed?

Reply:
Not needed, good spot thanks!